snapcraft/utils.py

```diff
@@ -29,10 +28,27 @@
 OSPlatform = namedtuple("OSPlatform", "system release machine")
 
 
-def is_managed_mode():
+def strtobool(value: str) -> bool:
+    """Convert a string representation of truth to true (1) or false (0).
+
+    :param value: a True value of 'y', 'yes', 't', 'true', 'on', and '1'
+        or a False value of 'n', 'no', 'f', 'false', 'off', and '0'.
+    :raises ValueError: if `value` is not a valid boolean value.
+    """
+    parsed_value = value.lower()
+
+    if parsed_value in ("y", "yes", "t", "true", "on", "1"):
+        return True
+    if parsed_value in ("n", "no", "f", "false", "off", "0"):
+        return False
+
+    raise ValueError(f"Invalid boolean value of {value!r}")
+
+
+def is_managed_mode() -> bool:
     """Check if snapcraft is running in a managed environment."""
     managed_flag = os.getenv("SNAPCRAFT_MANAGED_MODE", "n")
-    return distutils.util.strtobool(managed_flag) == 1
+    return strtobool(managed_flag)
```

Comment:
`SNAPCRAFT_MANAGED_MODE` is set by ourselves, can't we just assume we set it consistently with a known value?

Reply:
I guess, but we do use strtobool in the older code as well